authors/apps/articles/views.py

```diff
@@ -0,0 +1,109 @@
+import os
+import django
+import json
+import random
+
+from rest_framework import status, exceptions
+from rest_framework import generics
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from datetime import datetime, timedelta
+from rest_framework.views import Response
+from django.template.defaultfilters import slugify
+
+from .renderers import ArticleJSONRenderer
+from .models import Article
+from .serializers import ArticleSerializer
+from .messages import error_msgs, success_msg
+
+
+class ArticleAPIView(generics.ListCreateAPIView):
+    """
+        Article endpoints
+    """
+    # renderer_classes = (ArticleJSONRenderer,)
+    queryset = Article.objects.all()
+    serializer_class = ArticleSerializer
+
+    def post(self, request):
+        """
+            POST /api/v1/articles/
+        """
+        permission_classes = (IsAuthenticated,)
+        article = request.data.get("article", {})
+        context = {"request": request}
+        a_slug = slugify(request.data['article']['title'])
+        author = request.user.id
+        random_number = 1
+        unique_slug = a_slug
+        while Article.objects.filter(slug=unique_slug).exists():
+            unique_slug = '{}-{}'.format(a_slug, random_number)
+            random_number += 1
+        slug = unique_slug
+        article['slug'] = slug
+        serializer = self.serializer_class(data=article, context=context)
+        if serializer.is_valid():
+            serializer.save(author = request.user)
+            return Response(
+                serializer.data,
+                status=status.HTTP_201_CREATED
+            )
+        return Response(
+            serializer.errors,
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
+    def get(self, request):
+        """
+            GET /api/v1/articles/
+        """
+        permission_classes = (AllowAny,)
+        queryset = self.get_queryset()
+        serializer = ArticleSerializer(queryset, many=True)
+        return Response({
```

Comment:
```suggestion
       return Response({
            "articles": serializer.data,
            "articles_count": len(serializer.data)
        }, status=200)
```
also here add a response message

Reply:
Done by Wachira